src/v/rpc/transport.cc

```diff
@@ -200,50 +200,61 @@ transport::do_send(sequence_t seq, netbuf b, rpc::client_opts opts) {
             });
       });
 }
 
 void transport::dispatch_send() {
+    if (_is_sending_background) {
+        _has_skipped_dispatch = true;
+        return;
+    }
+
+    _is_sending_background = true;
+
     ssx::background
       = ssx::spawn_with_gate_then(_dispatch_gate, [this]() mutable {
             return ss::do_until(
-              [this] {
-                  return _requests_queue.empty()
-                         || _requests_queue.begin()->first
-                              > (_last_seq + sequence_t(1));
-              },
-              [this] {
-                  auto it = _requests_queue.begin();
-                  _last_seq = it->first;
-                  auto entry = std::move(it->second);
-                  _requests_queue.erase(it->first);
-                
-                  // `entry->resource_units` units are released once we are out of scope here
-                  // and that is intentional because the underlying write call
-                  // to the batched output stream guarantees us the in-order
-                  // delivery of the dispatched write calls, which is the intent
-                  // of holding on to the units up until this point.
+                     [this] {
+                         return _requests_queue.empty()
+                                || _requests_queue.begin()->first
+                                     > (_last_seq + sequence_t(1));
+                     },
+                     [this] {
+                         auto it = _requests_queue.begin();
+                         _last_seq = it->first;
+                         auto entry = std::move(it->second);
+                         _requests_queue.erase(it->first);
+
+                         // `entry->resource_units` units are released once we
+                         // are out of scope here and that is intentional
+                         // because the underlying write call to the batched
+                         // output stream guarantees us the in-order delivery of
+                         // the dispatched write calls, which is the intent of
+                         // holding on to the units up until this point.
 
-                  return ss::do_with(
-                    std::move(entry->buffer),
-                    [this](
-                      std::unique_ptr<netbuf>& buffer) {
-                        return std::move(*buffer).as_scattered().then(
-                          [this](auto v) {
-                              auto msg_size = v.size();
-                              return _out.write(std::move(v))
-                                .then([this, msg_size] {
-                                    _probe.add_bytes_sent(msg_size);
-                                });
-                          });
-                    });
+                         return ss::do_with(
+                                  std::move(entry->buffer),
+                                  [this](std::unique_ptr<netbuf>& buffer) {
```

Comment:
if this is pure code reformatting, please factor into a separate commit so it is easier to see what changes are being made.